src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -556,6 +540,216 @@ public void Add(NamespaceOrTypeSymbol symbol)
+                    case microsoft:
+                        if (containingNamespace.ContainingNamespace.IsGlobalNamespace && !_dictionary.ContainsKey(codeAnalysis))
+                        {
+                            Add(makeEmptyNamespace(codeAnalysis));
+                        }
+                        return;
+                    case compilerServices:
+                        if (isRuntime(containingNamespace.ContainingNamespace) && !_dictionary.ContainsKey("NonNullTypesAttribute"))
+                        {
+                            Add(InjectedNonNullTypesAttributeSymbol.Create(containingNamespace));
+                        }
+                        return;
+                    case codeAnalysis:
+                        if (isMicrosoft(containingNamespace.ContainingNamespace) && !_dictionary.ContainsKey("EmbeddedAttribute"))
+                        {
+                            Add(InjectedEmbeddedAttributeSymbol.Create(containingNamespace));
+                        }
+                        return;
+                }
+
+                InjectedNamespaceSymbol makeEmptyNamespace(string name)
+                {
+                    return new InjectedNamespaceSymbol(containingNamespace.ContainingModule, containingNamespace, name);
+                }
+
+                bool isSystem(NamespaceSymbol symbol)
+                {
+                    return symbol.Name == system && symbol.ContainingNamespace.IsGlobalNamespace;
+                }
+
+                bool isRuntime(NamespaceSymbol symbol)
+                {
+                    return symbol.Name == runtime && isSystem(symbol.ContainingNamespace);
+                }
+
+                bool isMicrosoft(NamespaceSymbol symbol)
+                {
+                    return symbol.Name == microsoft && symbol.ContainingNamespace.IsGlobalNamespace;
+                }
+            }
+
+            class InjectedNamespaceSymbol : NamespaceSymbol
+            {
+                private ModuleSymbol _module;
+                private NamespaceSymbol _containingNamespace;
+                private string _name;
+                private Dictionary<string, ImmutableArray<NamespaceOrTypeSymbol>> _nameToMembersMap;
+                private Dictionary<string, ImmutableArray<NamedTypeSymbol>> _nameToTypeMembersMap;
+                private ImmutableArray<Symbol> _lazyAllMembersOrdered;
+                private ImmutableArray<NamedTypeSymbol> _lazyTypeMembersOrdered;
+
+                public InjectedNamespaceSymbol(ModuleSymbol module, NamespaceSymbol containingNamespace, string name)
+                {
+                    Debug.Assert(name != null);
+                    _module = module;
+                    _containingNamespace = containingNamespace;
+                    _name = name;
+                }
+
+                public override string Name 
+                    => _name;
+
+                public override AssemblySymbol ContainingAssembly
```

Comment:
>public override AssemblySymbol ContainingAssembly [](start = 16, length = 49)

I think we can rely on implementation in base class.

Reply:
There is no implementation (just the abstract declaration).

---
In reply to: [216052712](https://github.com/dotnet/roslyn/pull/29180#discussion_r216052712) [](ancestors = 216052712)